# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.h
# Commit: a072d5ee2511
# Full Hash: a072d5ee251135b81716432bed6f5a7e185ad3c3
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-20 14:55:47
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 3: Implement the algorithm for obtaining tls and use it for wasm signal handling. r=lth
#   
#   This is the third part of series of patches to Frame without tls pointer.
#   Here we preserve initial tls in all entry stubs and then use it to find a proper tls instance for a given frame.
#   
# ==============================================================================

diff -r e8164759f391 -r a072d5ee2511 js/src/jit/MacroAssembler.h
--- a/js/src/jit/MacroAssembler.h	Tue Oct 20 10:24:19 2020 +0000
+++ b/js/src/jit/MacroAssembler.h	Tue Oct 20 10:24:32 2020 +0000
@@ -4420,6 +4420,9 @@
       : ABIArgIterBase<VecT, WasmABIArgGenerator>(types) {}
 };
 
+wasm::TlsData* ExtractCalleeTlsFromFrameWithTls(wasm::Frame* fp);
+wasm::TlsData* ExtractCallerTlsFromFrameWithTls(wasm::Frame* fp);
+
 }  // namespace jit
 }  // namespace js
 